elastic/kibana
# [Fleet] Update existing datastream lifecycle on package install/upgrade

## Summary

In #161905 we introduced support for DLM from package-spec in Fleet, those DLM are correctly installed for new datastream, but when updating an integration we need to update the existing datastream DLM, that PR does that.

Blocked on https://github.com/elastic/elasticsearch/pull/97732

## Tests

This is covered with unit tests

Manual test, there is no package with DLM yet you can use the elastic_agent package from that branch https://github.com/nchaulet/integrations/tree/test-dlm

* Start a fresh ES and Kibana
* install elastic_agent in 1.8.0 (without DLM)
* add a fleet server and check there is no DLM
* use a custom registry using the elastic_agent package with dlm install latest elastic_agent
* check the DLM policy is working for elastic_agent.fleet_server logs (data_retention is 2m)

## Todo 

- [x] Manual test instructions
